collect_app/src/test/java/org/odk/collect/android/widgets/GeoPointWidgetTest.java

```diff
@@ -1,27 +1,41 @@
 package org.odk.collect.android.widgets;
 
-import android.content.Intent;
+import android.view.View;
 
-import org.javarosa.core.model.QuestionDef;
 import org.javarosa.core.model.data.GeoPointData;
-import org.junit.Before;
+import org.javarosa.form.api.FormEntryPrompt;
 import org.junit.Test;
-import org.mockito.Mock;
+import org.junit.runner.RunWith;
 import org.odk.collect.android.R;
-import org.odk.collect.android.ShadowPlayServicesUtil;
-import org.odk.collect.android.activities.GeoPointActivity;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.widgets.base.BaseGeoWidgetTest;
-import org.robolectric.annotation.Config;
+import org.robolectric.RobolectricTestRunner;
 
-import androidx.annotation.NonNull;
-
-import static org.mockito.Mockito.when;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithReadOnly;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.widgetTestActivity;
 
 /**
  * @author James Knight
  */
 
+@RunWith(RobolectricTestRunner.class)
+public class GeoPointWidgetTest {
+
+    @Test
+    public void usingReadOnlyOption_makesAllClickableElementsDisabled() {
+        GeoPointWidget widget = createWidget(promptWithReadOnly());
+        assertThat(widget.findViewById(R.id.simple_button).getVisibility(), equalTo(View.GONE));
+    }
+
+    private GeoPointWidget createWidget(FormEntryPrompt prompt) {
+        return new GeoPointWidget(widgetTestActivity(), new QuestionDetails(prompt, "formAnalyticsID"));
+    }
+}
+
```

Comment:
Hi @seadowg! I had some doubts. As here, `GeoPointWidgetTest` does not inherit from any other tests. Should I go about it, or maintain the already implemented hierarchy. Also, I wanted to ask, is there any drawback when running tests having some hierarchy (in terms of running time, or some other factor)?

Reply:
I think we should go for the tests not having a hierarchy i.e. `GeoPointWidgetTest` should not extend anything.

As far as implementation goes I think it's a harder decision. I think a good way to start would be to rewrite the tests for these widgets (`GeoPointWidget`, `GeoShapeWidget` and `GeoTraceWidget`) as one PR. Then we can discuss how to rewrite the widgets using the new "framework".